### PR TITLE
CNDIT-792: Update validation logic to fix SFT and SPM unknown exception

### PR DESCRIPTION
**Pull request by ndduc01**
- Updated validation logic for it to work with SFT and SPM segment
- Updated time parse so it can take time zone into the account (specific for SPM)

 https://cdc-nbs.atlassian.net/browse/CNDIT-792?atlOrigin=eyJpIjoiNzQ5NzY5NTE1ZWQ2NGVkN2I1MTJjOGE0MmZmMDcyMGYiLCJwIjoiaiJ9

### CODE REVIEW

File: hl7-parser/src/main/java/gov/cdc/dataingestion/hl7/helper/integration/HL7Parser.java [hunk 1]
```diff
@@ -51,60 +50,29 @@ public String processFhsMessage(String message) {
...
-        } catch (HL7Exception e){
-            throw new DiHL7Exception(EX_MESSAGE + e.getMessage());
-        }
-
-        // Ignore sonar queue complain as this is coming from Library
-        ValidationRuleBuilder builder = new DefaultValidationBuilder() { // NOSONAR
-            @Override
-            protected  void configure() {
-                super.configure();
-                forAllVersions().message(ORU, ORU_01).onlyKnownSegments();
```
Comment (ndduc01) [inline hl7-parser/src/main/java/gov/cdc/dataingestion/hl7/helper/integration/HL7Parser.java:84]: this is the check kept failing the message with sft and spm segment

Comment (ndduc01) [reply hl7-parser/src/main/java/gov/cdc/dataingestion/hl7/helper/integration/HL7Parser.java:84]: perhaps it was the message she sent in at the time, but I was debugged this with her and able to reproduced it even in my local environment.

it had something to do with the escape characters '\r' and '\n', only in some specific scenario where input is appended with '\n' and di replace '\n' with '\r' (this some how cause the onlyKnowSegment check to not recognized the SFT and SPM segment

Comment (ndduc01) [reply hl7-parser/src/main/java/gov/cdc/dataingestion/hl7/helper/integration/HL7Parser.java:84]: also, additional info for this whenever it occurred in test.
I set the validation header back to false and the message was able to pass without throw the exception. This header basically force the message flow through the check above which in the failed ingestion

**Review by rshanmugam-enquizit — APPROVED** [review]
lgtm

**Pull request merged**